benchmarks/timeseries/benchmark_datasourcing.py

```diff
@@ -0,0 +1,116 @@
+import asyncio
+from time import perf_counter
+from typing import Tuple
+from unittest.mock import patch
+
+from frequenz.channels.util import Merge, MergeNamed
+from pytest_mock import MockerFixture
+
+from frequenz.sdk.actor import (
+    ChannelRegistry,
+    ComponentMetricRequest,
+    DataSourcingActor,
+)
+from frequenz.sdk.microgrid.client import Connection
+from frequenz.sdk.microgrid.component import (
+    Component,
+    ComponentCategory,
+    ComponentMetricId,
+)
+from tests.timeseries.mock_microgrid import MockMicrogrid
+
+
+@patch("frequenz.sdk.microgrid._microgrid._MICROGRID")
+async def benchmark_data_sourcing(
+    num_evc_chargers: int, num_msgs_per_battery: int, mock_microgrid: MockerFixture
+) -> None:
+    """Benchmark the data sourcing actor.
+
+    Args:
+        num_evc_chargers: number of EV Chargers to create for the mock microgrid.
+        num_msgs_per_battery: number of messages to send out for each battery.
+
+    Returns:
+        A tuple of (number of messages sent, time taken to stream the messages).
+    """
+    mock_grid = MockMicrogrid(
+        grid_side_meter=False,
+        num_values=num_msgs_per_battery,
+        sample_rate_s=0.00010,
+    )
+
+    mock_grid.add_ev_chargers(num_evc_chargers)
+
+    client = mock_grid.start_mock_client()
+    # Not working
+    # client = mock_grid.start_mock_client(mock_microgrid)
```

Comment:
As the mocker seems to be more like a context object, I thought I try it like this, with the patch decorator at the top, but passing this to the method also didn't work